packages/@lwc/synthetic-shadow/src/shared/node-collection-util.ts

```diff
@@ -16,16 +16,15 @@ import { isUndefined, isTrue } from './language';
  * Reference: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter#Polyfill
  */
 export function collectionFilter<T extends Node, K extends Element>(
-    this: NodeListOf<T> | HTMLCollectionOf<K>,
-    fn: (value: T | K, index?: number, collection?: NodeListOf<T> | HTMLCollectionOf<K>) => boolean,
-    thisArg?: any
+    collection: NodeListOf<T> | HTMLCollectionOf<K>,
+    fn: (value: T | K, index?: number, collection?: NodeListOf<T> | HTMLCollectionOf<K>) => boolean
 ): Array<T | K> {
     const res: Array<T | K> = [];
-    const length = this.length;
+    const length = collection.length;
     for (let i = 0; i < length; i++) {
-        const curr = this[i];
-        if (isTrue(fn.call(thisArg, curr, i, this))) {
-            res.push(curr);
+        const curr = collection[i];
+        if (isTrue(fn.call(undefined, curr, i, collection))) {
```

Comment:
if you're calling `fn.call()` with undefined context, then that's equivalent to do `fn(curr, i, collection)`, and you avoid the extra lookup for `.call`

Reply:
Yup, since we run in strict mode always. I think it is safe to use the `fn()` pattern.